### PR TITLE
VC issuance in Authz Code flow with considering scope parameter

### LINKED ISSUE #29725: VC issuance in Authz Code flow with considering “scope” parameter
### Description

The issue is a sub task of https://github.com/keycloak/keycloak/issues/26328 .

As mentioned in https://github.com/keycloak/keycloak/discussions/17616#discussioncomment-8627287 , the issue is for issuing a VC in Authz code flow with considering parameter.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by tnorimat**
closes #29725

### CODE REVIEW

**Comment by mposolda** [comment]
@wistefan @francis-pouatcha Could you please review this PR?

**Comment by mposolda** [comment]
@tnorimat Failing tests in `Keycloak CI / Base IT (6)` look like regression of the changes in this PR. Can you please doublecheck?

**Comment by tnorimat** [comment]
@mposolda Thank you for your comment.

> Error: OR] Failures: 
Error: OR] org.keycloak.testsuite.oid4vc.issuance.signing.OID4VCIssuerEndpointTest.testCredentialIssuanceWithAuthZCodeScopeUnmatched
Error: OR]   Run 1: OID4VCIssuerEndpointTest.testCredentialIssuanceWithAuthZCodeScopeUnmatched:586 expected:<400> but was:<500>
Error: OR]   Run 2: OID4VCIssuerEndpointTest.testCredentialIssuanceWithAuthZCodeScopeUnmatched:586 expected:<400> but was:<500>
Error: OR]   Run 3: OID4VCIssuerEndpointTest.testCredentialIssuanceWithAuthZCodeScopeUnmatched:586 expected:<400> but was:<500>

This failed test was newly added by this PR. In my environment this failure does not occur. I will investigate this failure.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oid4vc/issuance/signing/OID4VCIssuerEndpointTest.java [hunk 1]
```diff
@@ -458,7 +486,7 @@ public void testCredentialIssuanceWithAuthZCode() throws Exception {
 
             // 1. Get authoriZation code without scope specified by wallet
             // 2. Using the code to get accesstoken
-            String token = getBearerToken(oauth.openid(false).scope(null));
+            String token = getBearerToken(oauth.openid(false).scope("scope-test-credential"));
```
Comment (wistefan) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oid4vc/issuance/signing/OID4VCIssuerEndpointTest.java:489]: Wouldn´t it be better to create another test to ensure it works with and without the scope? The current test also does not match its description "Get authoriZation code without scope specified by wallet"

**Review by wistefan — CHANGES_REQUESTED** [review]

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oid4vc/issuance/signing/OID4VCIssuerEndpointTest.java (hunk 1 above)
Comment (tnorimat) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oid4vc/issuance/signing/OID4VCIssuerEndpointTest.java]: @wistefan Thank you for your comments. I prepared the following three types of test:
- Authz Code Flow, with scope, scope matching
- Authz Code Flow, with scope, scope un-matching
- Authz-Code Flow, without scope

**Comment by tnorimat** [comment]
@mposolda @wistefan I confirmed that the PR passed all CI test. If possible, could you check the PR?

**Comment by mposolda** [comment]
@tnorimat Thanks! I can merge if @wistefan also reviews and approves this PR.

**Review by wistefan — APPROVED** [review]
LGTM

**Comment by tnorimat** [comment]
@wistefan @mposolda Thank you for the review.

**Review by mposolda — APPROVED** [review]
@tnorimat @wistefan Thanks!

**Pull request merged**